src/main/java/com/conveyal/datatools/manager/auth/Auth0Connection.java

```diff
@@ -60,17 +77,64 @@ public static void checkUser(Request req) {
         // Validate the JWT and cast into the user profile, which will be attached as an attribute on the request object
         // for downstream controllers to check permissions.
         try {
-            byte[] decodedSecret = new Base64().decode(getConfigPropertyAsText("AUTH0_SECRET"));
-            JWTVerifier verifier = new JWTVerifier(decodedSecret);
-            Map<String, Object> jwt = verifier.verify(token);
+            Map<String, Object> jwt = verifyToken(token);
+            remapTokenValues(jwt);
             Auth0UserProfile profile = MAPPER.convertValue(jwt, Auth0UserProfile.class);
+            // The user attribute is used on the server side to check user permissions and does not have all of the
+            // fields that the raw Auth0 profile string does.
             req.attribute("user", profile);
+            // The raw_user attribute is used to return the complete user profile string to the client upon request in
+            // the UserController. Previously the client sought the profile directly from Auth0 but the need to add
+            req.attribute("raw_user", MAPPER.writeValueAsString(jwt));
         } catch (Exception e) {
             LOG.warn("Login failed to verify with our authorization provider.", e);
             haltWithMessage(401, "Could not verify user's token");
         }
     }
 
+    /**
+     * Choose the correct JWT verification algorithm (based on the values present in env.yml config) and verify the JWT
+     * token.
+     */
+    private static Map<String, Object> verifyToken(String token) throws NoSuchAlgorithmException, NoSuchProviderException, InvalidKeySpecException, IOException, JWTVerifyException, InvalidKeyException, SignatureException {
+        if (verifier == null) {
+            if (hasConfigProperty("AUTH0_SECRET")) {
+                // Use HS256 algorithm to verify token (uses client secret).
+                byte[] decodedSecret = new org.apache.commons.codec.binary.Base64().decode(getConfigPropertyAsText("AUTH0_SECRET"));
+                verifier = new JWTVerifier(decodedSecret);
+            } else if (hasConfigProperty("AUTH0_PUBLIC_KEY")) {
+                // Use RS256 algorithm to verify token (uses public key/.pem file).
+                PublicKey publicKey = PemReader.readPublicKey(getConfigPropertyAsText("AUTH0_PUBLIC_KEY"));
+                verifier = new JWTVerifier(publicKey);
+            } else {
+                throw new RuntimeException("Server authentication provider not configured correctly.");
+            }
+        }
```

Comment:
I think it'd be a good idea to wrap the creation of the verifier in a try/catch and log accordingly if there was an issue doing so.  Otherwise, this error will just get swallowed up in the catch block in the `checkUser` method.  Once that is done, this method could be renamed `getVerifier`.